crypto/encode_decode/decoder_lib.c

```diff
@@ -512,20 +514,32 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
     int err, ok = 0;
     /* For recursions */
     struct decoder_process_data_st new_data;
-    const char *object_type = NULL;
+    const char *data_type = NULL;
+    const char *data_structure = NULL;
 
     memset(&new_data, 0, sizeof(new_data));
     new_data.ctx = data->ctx;
+    new_data.recursion = data->recursion + 1;
+
+#define LEVEL_STR ">>>>>>>>>>>>>>>>"
+#define LEVEL (&LEVEL_STR[sizeof(LEVEL_STR) - new_data.recursion - 1])
```

Comment:
I know this is only relevant if we are tracing - but shouldn't we handle the case where `new_data.recursion >= sizeof(LEVEL_STR)`?

Reply:
There is a limit to the amount of levels: 10

Reply:
Where is that implemented? Perhaps a comment here?

Reply:
Hmmmm, it is of course possible for the caller to add decoders at will.  So ok, I recoded the macro for the case you mention.